test/SlowTests/Server/Documents/ETL/EtlTimeSeriesTests.cs

```diff
@@ -0,0 +1,2480 @@
+        )
+        {
+            var src = GetDocumentStore(_options);
+
+            try
+            {
+                var time = new DateTime(2020, 04, 27);
+                const string timeSeriesName = "Heartrate";
+                const string tag = "fitbit";
+                const double value = 58d;
+                const string documentId = "users/1";
+
+                using (var session = src.OpenAsyncSession())
+                {
+                    var entity = new User {Name = "Joe Doe"};
+                    await session.StoreAsync(entity, documentId);
+                    session.TimeSeriesFor(documentId, timeSeriesName).Append(time, new[] {value}, tag);
+
+                    await session.SaveChangesAsync();
+                }
+
+                var database = GetDatabase(src.Database).Result;
+                using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                {
+                    var testRavenEtlScript = new TestRavenEtlScript
+                    {
+                        DocumentId = documentId,
+                        Configuration = new RavenEtlConfiguration()
+                        {
+                            Name = "simulate", 
+                            Transforms =
+                            {
+                                new Transformation
+                                {
+                                    Collections = collections.ToList(), 
+                                    Name = "OrdersAndLines", 
+                                    Script = script, ApplyToAllDocuments = collections.Length == 0
+                                }
+                            },
+                        },
+                    };
+                    var result = (RavenEtlTestScriptResult)RavenEtl.TestScript(testRavenEtlScript, database, database.ServerStore, context);
+ 
+                    Assert.Equal(0, result.TransformationErrors.Count);
+
+                    var timeSeriesCommand = result.Commands.OfType<TimeSeriesBatchCommandData>().FirstOrDefault(c => 
+                        c.Name == timeSeriesName && c.TimeSeries?.Appends != null && c.TimeSeries.Appends.Any());
+                    
+                    Assert.NotNull(timeSeriesCommand);
+                    Assert.Equal(CommandType.TimeSeries, timeSeriesCommand.Type);
+                    Assert.True(timeSeriesCommand.FromEtl);
+                    
+                    var actualSent = timeSeriesCommand.TimeSeries.Appends.FirstOrDefault();
+                    Assert.NotNull(actualSent);
+                    Assert.Equal(tag, actualSent.Tag);
+                    Assert.Equal(time, actualSent.Timestamp);
+                    Assert.Equal(value, actualSent.Values.FirstOrDefault());
+                }
+            }
+            catch (Exception)
+            {
+                if (shouldSuccess)
+                    throw;
+
+                shouldSuccess = true;
```

Comment:
I don't like this try catch on entire method body. Can we do better? If we have `shouldSuccess: false` then shouldn't it throw on _specific_ assert or operation. I think we should handle it there instead of generic `try / catch`